flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/BinaryRow.java

```diff
@@ -129,13 +130,15 @@ public int getArity() {
 	}
 
 	@Override
-	public byte getHeader() {
+	public RowKind getRowKind() {
 		// first nullBitsSizeInBytes byte is header.
-		return segments[0].get(offset);
+		byte header = segments[0].get(offset);
+		return RowKind.values()[header];
```

Comment:
IMHO, using enum's values and ordinal is not very elegant.
How about adding a field explicitly whose meaning is byte encoding for each enum?